presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -278,7 +278,7 @@ public QueryInfo getQueryInfo(StageInfo rootStage)
                 }
 
                 PlanFragment plan = stageInfo.getPlan();
-                if (plan != null && plan.getPartitionedSourceNode() instanceof TableScanNode) {
+                if (plan != null && plan.getPartitionedSourceNodes().stream().anyMatch(TableScanNode.class::isInstance)) {
```

Comment:
What type could this be other than `TableScanNode`? I followed the data path and don't see how it can be any other type.